### PR TITLE
When removing traitor role, auto remove the uplink

### PR DESCRIPTION
**Pull request by Miraviel**
## What Does This PR Do

When losing the traitor status, the uplink also gets deleted.

## Why It's Good For The Game

When admins de-traitor people, they need to remove their uplink as well by pressing a second button. This should be all automated, birdbrains like me might forget to take the extra step.

The "Uplink - take" button does not clear the pda/headset's `hidden_uplink` reference either.

## Images of changes

Delete PDA uplink:

https://user-images.githubusercontent.com/33333517/189718298-a5df3ce3-bd48-4419-abfe-b154a62d7935.mp4

Delete headset uplink:

https://user-images.githubusercontent.com/33333517/189719263-0fbd6fd9-c89b-460f-997b-8bd927c72d79.mp4

Delete implanter uplink:

https://user-images.githubusercontent.com/33333517/189719550-55fc12dc-1173-4a56-930f-6589d5fef756.mp4

## Testing

By doing the videos above.

## Changelog
:cl:
tweak: De-traitoring someone (via admin tools) now automatically removes their uplink.
/:cl:

### CODE REVIEW

**Comment by Contrabang** [comment]
Does this work if the uplinks are stored in their bag? Like putting your PDA in your bag(or a box in your bag) then detraitoring yourself seems like it might not work. If it is like this I recommend looking into how cultists are able to summon their dagger from anywhere in their bag, I forget the exact name of the proc.

**Comment by Komrad822** [comment]
Ah.... yes... this will help prevent the accidental "All the starting cultist have tot items and one of them is the hop...." situation from happening again... possibly...

**Comment by Miraviel** [comment]
> Ah.... yes... this will help prevent the accidental "All the starting cultist have tot items and one of them is the hop...." situation from happening again... possibly...

Never heard of this issue, nor I really see how it pertains to the current PR?

**Comment by Komrad822** [comment]
>Never heard of this issue, nor I really see how it pertains to the current PR?

Not really an issue, but round wasnt set to secret, tot round was auto called out. Admins changed all tots to cultist but the uplinks weren't auto-removed, thus you had cultist with carp etc. IDK if it was intentional or not, but I do know cultist were told to go easy. 

TLDR: PR good. Wanted to make a comment that ~~possibly~~ pertains to PR without having to monologue about the exact situation.

**Review by Sirryan2002 — APPROVED** [review]
code looks fine

**Review by hal9000PR — APPROVED** [review]
